beam-postgres/Database/Beam/Postgres/Syntax.hs

```diff
@@ -1371,6 +1421,23 @@ pgDropExtensionSyntax extName =
 --                                        Nothing
 --      buildJoinFrom tbl newSource Nothing
 
+
+-- TODO
+-- Don't directly expose sum-type constructors?
+-- Constrain 'table' to exist within the query (using s maybe?)
+-- Constrain Q to not have aggregations (at least the locked tables I think, using s?)
+--
+--   https://www.postgresql.org/docs/10/static/sql-select.html#SQL-FOR-UPDATE-SHARE
+--   "The locking clauses cannot be used in contexts where returned rows cannot be clearly
+--   identified with individual table rows; for example they cannot be used with aggregation."
```

Comment:
Oh this is an interesting restriction. I'll have to think what the best way to deal with this one is. We could probably do it with the `s` parameter, but may need to make some changes in `beam-core`.